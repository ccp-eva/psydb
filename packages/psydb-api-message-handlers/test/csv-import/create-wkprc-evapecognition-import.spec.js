'use strict';
var { ejson, omit } = require('@mpieva/psydb-core-utils');
var { expect } = require('@mpieva/psydb-api-mocha-test-tools/chai');
var { ObjectId } = require('@cdxoo/mongo-test-helpers');
var { getContent: loadCSV } = require('@mpieva/psydb-fixtures/csv');

var jsonify = (that) => (
    JSON.parse(JSON.stringify(that))
);

var RootHandler = require('../src/');

var omitNonsense = ({ from }) => omit({
    from, paths: [ '_id', '_rohrpostMetadata' ]
});

describe('csv-import/create-subject-import fs-malaysia', function () {
    var db, sendMessage, fileId;
    beforeEach(async function () {
        await this.restore('2023-11-29__0517-wkprc-fieldsite');
        
        db = this.getDbHandle();
        ([ sendMessage ] = this.createMessenger({
            RootHandler,
            ...(await this.createFakeLogin({ email: 'root@example.com' }))
        }));

        var file = await createFakeFileUpload({
            db, buffer: loadCSV('evapecognition/simple'),
        });
        fileId = file._id;
    });

    it('does the thing', async function () {
        var studyId = ObjectId("6566b5c26c830cb226c1389b");
        var locationId = ObjectId("64d42de0443aa279ca4cb2e8");
        var subjectType = 'wkprc_chimpanzee';
        var labOperatorIds = [
            ObjectId("64d42ddf443aa279ca4cb2c9"),
            ObjectId("64d42ddf443aa279ca4cb2c5"),
        ];

        var koaContext = await sendMessage({
            type: 'csv-import/experiment/create-wkprc-evapecognition',
            timezone: 'UTC',
            payload: jsonify({
                subjectType,
                studyId,
                locationId,
                fileId,
                labOperatorIds
            })
        });

        var { body } = koaContext.response;
        var [ ageFrameUpdate ] = body.data;

        console.log(ageFrameUpdate);

        var record = omitNonsense({
            from: await this.getRecord('ageFrame', {
                _id: ageFrameUpdate.channelId
            })
        });

        console.dir(ejson(record), { depth: null });
        expect(ejson(record)).toMatchSnapshot();
    })
})
