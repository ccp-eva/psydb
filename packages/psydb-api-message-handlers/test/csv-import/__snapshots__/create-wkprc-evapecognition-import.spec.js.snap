// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csv-import/experiment/create-wkprc-evapecognition does the thing 1`] = `
Array [
  Object {
    "realType": "apestudies-wkprc-default",
    "state": Object {
      "color": "",
      "experimentName": "choice_task",
      "experimentOperatorIds": Array [
        Object {
          "$oid": "64d42ddf443aa279ca4cb2c9",
        },
        Object {
          "$oid": "64d42ddf443aa279ca4cb2c5",
        },
      ],
      "experimentOperatorTeamId": null,
      "generalComment": "",
      "interval": Object {
        "end": Object {
          "$date": "2023-03-22T11:00:00.000Z",
        },
        "start": Object {
          "$date": "2023-03-22T11:00:00.000Z",
        },
      },
      "isCanceled": false,
      "isPostprocessed": true,
      "locationId": Object {
        "$oid": "64d42de0443aa279ca4cb2e8",
      },
      "locationRecordType": "wkprc_ape_location",
      "roomOrEnclosure": "Observation Room",
      "selectedSubjectGroupIds": Array [],
      "selectedSubjectIds": Array [
        Object {
          "$oid": "64d42de0443aa279ca4cb2fd",
        },
      ],
      "studyId": Object {
        "$oid": "6566b5c26c830cb226c1389b",
      },
      "studyRecordType": "wkprc_study",
      "subjectData": Array [
        Object {
          "comment": "focal_participant; Some Comment",
          "excludeFromMoreExperimentsInStudy": false,
          "invitationStatus": "scheduled",
          "participationStatus": "participated",
          "subjectId": Object {
            "$oid": "64d42de0443aa279ca4cb2fd",
          },
        },
      ],
      "subjectGroupId": Object {
        "$oid": "64d42de0443aa279ca4cb2ee",
      },
      "timezone": "Europe/Berlin",
    },
    "type": "manual",
  },
  Object {
    "realType": "apestudies-wkprc-default",
    "state": Object {
      "color": "",
      "experimentName": "choice_task",
      "experimentOperatorIds": Array [
        Object {
          "$oid": "64d42ddf443aa279ca4cb2c9",
        },
        Object {
          "$oid": "64d42ddf443aa279ca4cb2c5",
        },
      ],
      "experimentOperatorTeamId": null,
      "generalComment": "",
      "interval": Object {
        "end": Object {
          "$date": "2023-03-23T11:00:00.000Z",
        },
        "start": Object {
          "$date": "2023-03-23T11:00:00.000Z",
        },
      },
      "isCanceled": false,
      "isPostprocessed": true,
      "locationId": Object {
        "$oid": "64d42de0443aa279ca4cb2e8",
      },
      "locationRecordType": "wkprc_ape_location",
      "roomOrEnclosure": "Observation Room",
      "selectedSubjectGroupIds": Array [],
      "selectedSubjectIds": Array [
        Object {
          "$oid": "64d42de0443aa279ca4cb2fd",
        },
      ],
      "studyId": Object {
        "$oid": "6566b5c26c830cb226c1389b",
      },
      "studyRecordType": "wkprc_study",
      "subjectData": Array [
        Object {
          "comment": "focal_participant; Some Comment",
          "excludeFromMoreExperimentsInStudy": false,
          "invitationStatus": "scheduled",
          "participationStatus": "participated",
          "subjectId": Object {
            "$oid": "64d42de0443aa279ca4cb2fd",
          },
        },
      ],
      "subjectGroupId": Object {
        "$oid": "64d42de0443aa279ca4cb2ee",
      },
      "timezone": "Europe/Berlin",
    },
    "type": "manual",
  },
  Object {
    "realType": "apestudies-wkprc-default",
    "state": Object {
      "color": "",
      "experimentName": "choice_task",
      "experimentOperatorIds": Array [
        Object {
          "$oid": "64d42ddf443aa279ca4cb2c9",
        },
        Object {
          "$oid": "64d42ddf443aa279ca4cb2c5",
        },
      ],
      "experimentOperatorTeamId": null,
      "generalComment": "",
      "interval": Object {
        "end": Object {
          "$date": "2023-03-24T11:00:00.000Z",
        },
        "start": Object {
          "$date": "2023-03-24T11:00:00.000Z",
        },
      },
      "isCanceled": false,
      "isPostprocessed": true,
      "locationId": Object {
        "$oid": "64d42de0443aa279ca4cb2e8",
      },
      "locationRecordType": "wkprc_ape_location",
      "roomOrEnclosure": "Outdoor Enclosure",
      "selectedSubjectGroupIds": Array [],
      "selectedSubjectIds": Array [
        Object {
          "$oid": "64d42de0443aa279ca4cb2fd",
        },
        Object {
          "$oid": "64d42de0443aa279ca4cb301",
        },
      ],
      "studyId": Object {
        "$oid": "6566b5c26c830cb226c1389b",
      },
      "studyRecordType": "wkprc_study",
      "subjectData": Array [
        Object {
          "comment": "partner; Some Comment",
          "excludeFromMoreExperimentsInStudy": false,
          "invitationStatus": "scheduled",
          "participationStatus": "participated",
          "subjectId": Object {
            "$oid": "64d42de0443aa279ca4cb2fd",
          },
        },
        Object {
          "comment": "stooge; Some Comment",
          "excludeFromMoreExperimentsInStudy": false,
          "invitationStatus": "scheduled",
          "participationStatus": "participated",
          "subjectId": Object {
            "$oid": "64d42de0443aa279ca4cb301",
          },
        },
      ],
      "subjectGroupId": Object {
        "$oid": "64d42de0443aa279ca4cb2ee",
      },
      "timezone": "Europe/Berlin",
    },
    "type": "manual",
  },
]
`;

exports[`csv-import/experiment/create-wkprc-evapecognition does the thing 2`] = `
Array [
  Object {
    "_P": Object {
      "excludeFromMoreExperimentsInStudy": false,
      "realType": "apestudies-wkprc-default",
      "status": "participated",
      "studyId": Object {
        "$oid": "6566b5c26c830cb226c1389b",
      },
      "studyType": "wkprc_study",
      "timestamp": Object {
        "$date": "2023-03-22T11:00:00.000Z",
      },
      "timezone": "Europe/Berlin",
      "type": "manual",
    },
    "_id": Object {
      "$oid": "64d42de0443aa279ca4cb2fd",
    },
  },
  Object {
    "_P": Object {
      "excludeFromMoreExperimentsInStudy": false,
      "realType": "apestudies-wkprc-default",
      "status": "participated",
      "studyId": Object {
        "$oid": "6566b5c26c830cb226c1389b",
      },
      "studyType": "wkprc_study",
      "timestamp": Object {
        "$date": "2023-03-23T11:00:00.000Z",
      },
      "timezone": "Europe/Berlin",
      "type": "manual",
    },
    "_id": Object {
      "$oid": "64d42de0443aa279ca4cb2fd",
    },
  },
  Object {
    "_P": Object {
      "excludeFromMoreExperimentsInStudy": false,
      "realType": "apestudies-wkprc-default",
      "status": "participated",
      "studyId": Object {
        "$oid": "6566b5c26c830cb226c1389b",
      },
      "studyType": "wkprc_study",
      "timestamp": Object {
        "$date": "2023-03-24T11:00:00.000Z",
      },
      "timezone": "Europe/Berlin",
      "type": "manual",
    },
    "_id": Object {
      "$oid": "64d42de0443aa279ca4cb2fd",
    },
  },
  Object {
    "_P": Object {
      "excludeFromMoreExperimentsInStudy": false,
      "realType": "apestudies-wkprc-default",
      "status": "participated",
      "studyId": Object {
        "$oid": "6566b5c26c830cb226c1389b",
      },
      "studyType": "wkprc_study",
      "timestamp": Object {
        "$date": "2023-03-24T11:00:00.000Z",
      },
      "timezone": "Europe/Berlin",
      "type": "manual",
    },
    "_id": Object {
      "$oid": "64d42de0443aa279ca4cb301",
    },
  },
]
`;
